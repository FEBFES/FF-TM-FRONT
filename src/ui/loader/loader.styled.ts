import styled, { keyframes } from 'styled-components';

export const rotate = keyframes`
  0% {
    transform: rotate(0);
  }
  100% {
    transform: rotate(360deg);
  }
`;

export const SLoaderContainer = styled.div`
  margin-left: 5px;
  animation: 1.4s ease-in-out ${rotate} infinite;
`;
