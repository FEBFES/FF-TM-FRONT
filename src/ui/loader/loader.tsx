import React from 'react';
import { faSpinner } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { SizeProp } from '@fortawesome/fontawesome-svg-core';
import { SLoaderContainer } from './loader.styled';

interface LoaderProps {
  size?: SizeProp;
}

export const Loader: React.FC<LoaderProps> = ({ size }): JSX.Element => {
  return (
    <SLoaderContainer>
      <FontAwesomeIcon size={size} icon={faSpinner} />
    </SLoaderContainer>
  );
};
