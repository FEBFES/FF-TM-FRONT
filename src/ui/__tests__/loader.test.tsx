import React from 'react';
import { Loader } from '../loader/loader';
import renderer from 'react-test-renderer';

it('SNAPSHOTS: UI - Loader component - render', () => {
  const elem = renderer
    .create(
      <div>
        <Loader size="1x" />
      </div>
    )
    .toJSON();
  expect(elem).toMatchSnapshot();
});
