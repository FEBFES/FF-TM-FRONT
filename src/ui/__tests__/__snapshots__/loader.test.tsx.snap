// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SNAPSHOTS: UI - Loader component - render 1`] = `
<div>
  <div
    className="sc-bcPKhP haSZKe"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-spinner fa-1x "
      data-icon="spinner"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M304 48c0-26.5-21.5-48-48-48s-48 21.5-48 48s21.5 48 48 48s48-21.5 48-48zm0 416c0-26.5-21.5-48-48-48s-48 21.5-48 48s21.5 48 48 48s48-21.5 48-48zM48 304c26.5 0 48-21.5 48-48s-21.5-48-48-48s-48 21.5-48 48s21.5 48 48 48zm464-48c0-26.5-21.5-48-48-48s-48 21.5-48 48s21.5 48 48 48s48-21.5 48-48zM142.9 437c18.7-18.7 18.7-49.1 0-67.9s-49.1-18.7-67.9 0s-18.7 49.1 0 67.9s49.1 18.7 67.9 0zm0-294.2c18.7-18.7 18.7-49.1 0-67.9S93.7 56.2 75 75s-18.7 49.1 0 67.9s49.1 18.7 67.9 0zM369.1 437c18.7 18.7 49.1 18.7 67.9 0s18.7-49.1 0-67.9s-49.1-18.7-67.9 0s-18.7 49.1 0 67.9z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;
